### PR TITLE
wireshark: fix build on x86_64-darwin

### PR DESCRIPTION
**Pull request by wegank**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
@ofborg build wireshark wireshark.passthru.tests

**Comment by wegank** [comment]
Wait, what? How is `buildPackage` used in `darwin.apple_sdk_11_0.callPackage` before this PR?

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -30941,8 +30941,8 @@ with pkgs;
 
   welle-io = libsForQt5.callPackage ../applications/radio/welle-io { };
 
-  wireshark = callPackage ../applications/networking/sniffers/wireshark {
-    inherit (darwin.apple_sdk.frameworks) ApplicationServices SystemConfiguration;
+  wireshark = darwin.apple_sdk_11_0.callPackage ../applications/networking/sniffers/wireshark {
```
Comment (fpletz) [inline pkgs/top-level/all-packages.nix:30944]: Why is it necessary to use a custom `callPackage` for darwin support?

Comment (wegank) [reply pkgs/top-level/all-packages.nix:30944]: The point is that `x86_64-darwin` targets macOS 10.12, which is too old for Wireshark, so the macOS 11 SDK needs to be included somehow. `qt6Packages.callPackage` should also work here.

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 238914` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>dbmonster</li>
    <li>ostinato</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python311Packages.dissect</li>
    <li>python311Packages.dissect.dist</li>
  </ul>
</details>
<details>
  <summary>19 packages built:</summary>
  <ul>
    <li>credslayer</li>
    <li>credslayer.dist</li>
    <li>hfinger</li>
    <li>hfinger.dist</li>
    <li>python310Packages.dissect</li>
    <li>python310Packages.dissect-cobaltstrike</li>
    <li>python310Packages.dissect-cobaltstrike.dist</li>
    <li>python310Packages.dissect.dist</li>
    <li>python310Packages.pyshark</li>
    <li>python310Packages.pyshark.dist</li>
    <li>python311Packages.dissect-cobaltstrike</li>
    <li>python311Packages.dissect-cobaltstrike.dist</li>
    <li>python311Packages.pyshark</li>
    <li>python311Packages.pyshark.dist</li>
    <li>termshark</li>
    <li>tshark (wireshark-cli)</li>
    <li>tshark.dev (wireshark-cli.dev)</li>
    <li>wireshark (wireshark-qt)</li>
    <li>wireshark.dev (wireshark-qt.dev)</li>
  </ul>
</details>

**Pull request merged**